As i see, the missing dependency is now included in the newly created 
`distribution/feature-packs/server-feature-pack-dependencies/pom.xml`
thus my fix i would say is now included in the master branch.
However, on a fresh install (i.e. git clone)  and performing a 
`mvn -Pdistribution -pl distribution/server-dist -am -Dmaven.test.skip clean install `
the build now fails with the following build summary:

Reactor Summary for Keycloak 15.0.0-SNAPSHOT:
[INFO] 
[INFO] Keycloak ........................................... SUCCESS [  0.538 s]
[INFO] Keycloak Common .................................... SUCCESS [  2.163 s]
[INFO] Keycloak Core ...................................... SUCCESS [  1.154 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  1.261 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  1.993 s]
[INFO] Keycloak Kerberos Federation ....................... SUCCESS [  0.172 s]
[INFO] Keycloak LDAP UserStoreProvider .................... SUCCESS [  0.577 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.523 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  0.800 s]
[INFO] Keycloak REST Services ............................. SUCCESS [  5.828 s]
[INFO] Keycloak JS Integration ............................ SUCCESS [  1.814 s]
[INFO] Keycloak Themes .................................... SUCCESS [01:05 min]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.009 s]
[INFO] Keycloak Model JPA ................................. SUCCESS [  1.141 s]
[INFO] Keycloak Model Naive Map ........................... SUCCESS [  0.988 s]
[INFO] Keycloak Model Infinispan .......................... SUCCESS [  1.842 s]
[INFO] Keycloak SSSD Federation ........................... SUCCESS [  0.514 s]
[INFO] KeyCloak Authz: Parent ............................. SUCCESS [  0.011 s]
[INFO] KeyCloak AuthZ: Provider Parent .................... SUCCESS [  0.008 s]
[INFO] KeyCloak AuthZ: Common Policy Providers ............ SUCCESS [  0.213 s]
[INFO] Keycloak WildFly Integration ....................... SUCCESS [  0.008 s]
[INFO] Keycloak WildFly Add User Script ................... SUCCESS [  0.195 s]
[INFO] Keycloak WildFly Extensions ........................ SUCCESS [  0.388 s]
[INFO] Keycloak WildFly Server Subsystem .................. SUCCESS [  0.652 s]
[INFO] Keycloak Integration ............................... SUCCESS [  0.014 s]
[INFO] Keycloak Client CLI ................................ SUCCESS [  0.008 s]
[INFO] Keycloak Client Registration CLI ................... SUCCESS [  1.629 s]
[INFO] Keycloak Admin CLI ................................. SUCCESS [  1.380 s]
[INFO] Keycloak Client CLI Distribution ................... SUCCESS [  0.789 s]
[INFO] Distribution Parent ................................ SUCCESS [  0.008 s]
[INFO] Feature Pack Builds ................................ SUCCESS [  0.007 s]
[INFO] Keycloak Feature Pack: Server Dependencies ......... SUCCESS [  0.012 s]
[INFO] Galleon Feature Pack Builds ........................ SUCCESS [  0.007 s]
[INFO] Keycloak Galleon Feature Pack: Server .............. SUCCESS [ 26.231 s]
[INFO] Keycloak Server Galleon Based Distribution ......... FAILURE [ 21.458 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:21 min
[INFO] Finished at: 2021-07-05T10:56:33+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:unpack (unpack-and-inject-keycloak-server-feature-pack-licenses) on project keycloak-server-dist: Unable to find/resolve artifact.: Could not find artifact org.keycloak:keycloak-server-feature-pack:zip:15.0.0-SNAPSHOT -> [Help 1]

I saw that now, the **org.keycloak:keycloak-server-feature-pack** was not included in the **org.keycloak:server-galleon-pack**.
Adding it, seems to resolve the new bug.
Here you are.

--- a/distribution/galleon-feature-packs/server-galleon-pack/pom.xml
+++ b/distribution/galleon-feature-packs/server-galleon-pack/pom.xml
@@ -102,6 +102,20 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-server-feature-pack</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
         <!-- module and copy artifact dependencies -->
         <dependency>
             <groupId>org.keycloak</groupId>

